scripts/cargo-clippy-nightly.sh

```diff
@@ -24,4 +24,9 @@ source "$here/../ci/rust-version.sh" nightly
 #   ref: https://github.com/rust-lang/rust/issues/66287
 "$here/cargo-for-all-lock-files.sh" -- \
   "+${rust_nightly}" clippy \
-  --workspace --all-targets --features dummy-for-ci-check,frozen-abi
+  --workspace --all-targets --features dummy-for-ci-check,frozen-abi -- \
+  --deny=warnings \
```

Comment:
when we had these in clippy.toml, did they apply to all workspaces? agreed that putting them in Cargo.toml is pretty unmaintainable so long as we have multiple workspaces in the monorepo

Reply:
Keeping them in sync is a matter of a grep checking for relevant section in cargo.toml, no?

Reply:
if I don't miss anything, clippy.toml couldn't do that. for allowing/denying lints, we can only do either
- embedded in code
- cli flags
- Cargo.toml

ref: https://doc.rust-lang.org/clippy/configuration.html#allowingdenying-lints

Reply:
yes this is correct, we either do CLI flags (which is not ideal as it prevents simply running `cargo +nightly clippy` during development as you have to call the script which will set all the CLAs for you) or we make sure crates inherit main config correctly. I'd prefer crates inheriting correctly from workspace.

Reply:
```bash
files=`find . -name "Cargo.toml" `
for file in $files; do
    if ! grep -q "^\[lints\]" "$file"; then
      echo "Lints missing in $file"
      #echo -e "\n[lints]\nworkspace = true" >> "$file"
      #echo "Added [lints] to $file"
      # maybe call cargo sort here?
    else
      echo "$file already has [lints]"
    fi
  done
```

Reply:
I prefer not to rely on Cargo.toml if we don’t have CI to ensure the rules are consistent in this repo and to enforce that newly added crates include lints.workspace = true

Reply:
I’m happy to do improvements in separate PRs. for this PR, I would like to focus on reverting the lint checks back to how they were before.

Reply:
> Keeping them in sync is a matter of a grep checking for relevant section in cargo.toml, no?

we want fewer, not more, rube goldberg machines